tools/build_defs/repo/http.bzl

```diff
@@ -136,25 +133,21 @@ def _http_archive_impl(ctx):
     if ctx.attr.build_file and ctx.attr.build_file_content:
         fail("Only one of build_file and build_file_content can be provided.")
 
-    all_urls = _get_all_urls(ctx)
-    auth = get_auth(ctx, all_urls)
-
+    source_urls = _get_source_urls(ctx)
     download_info = ctx.download_and_extract(
-        # TODO(fzakaria): all_urls here has the remote_patch URL which is incorrect
-        # I believe this to be a file
-        all_urls,
+        source_urls,
         ctx.attr.add_prefix,
         ctx.attr.sha256,
         ctx.attr.type,
         ctx.attr.strip_prefix,
-        canonical_id = ctx.attr.canonical_id or get_default_canonical_id(ctx, all_urls),
-        auth = auth,
+        canonical_id = ctx.attr.canonical_id or get_default_canonical_id(ctx, source_urls),
```

Comment:
@fmeum Should we also apply `get_default_canonical_id` to other downloads?

Reply:
It makes sense to apply to everything with user-specified URLs. Remote files and patches are a difficult case: They could theoretically be user-specified, but in practice come from registries and thus aren't user-specified. I would lean towards *not* setting the id for these downloads.

Reply:
SGTM